test/fixtures/azure-fixture/src/main/java/fixture/azure/AzureHttpHandler.java

```diff
@@ -145,83 +148,115 @@ public void handle(final HttpExchange exchange) throws IOException {
+                final MockAzureBlobStore.BlobType blobType = MockAzureBlobStore.BlobType.fromXMSBlobType(blobTypeHeader);
+                if (blobType == null) {
+                    throw new MockAzureBlobStore.ConflictException("InvalidHeaderValue", "Unable to parse blobType: " + blobTypeHeader);
                 }
+                final String ifNoneMatch = exchange.getRequestHeaders().getFirst("If-None-Match");
+                mockAzureBlobStore.putBlob(
+                    blobPath(exchange),
+                    Streams.readFully(exchange.getRequestBody()),
+                    blobType,
+                    ifNoneMatch,
+                    leaseId(exchange)
+                );
                 exchange.getResponseHeaders().add("x-ms-request-server-encrypted", "false");
                 exchange.sendResponseHeaders(RestStatus.CREATED.getStatus(), -1);
 
             } else if (Regex.simpleMatch("HEAD /" + account + "/" + container + "/*", request)) {
                 // Get Blob Properties (see https://docs.microsoft.com/en-us/rest/api/storageservices/get-blob-properties)
-                final BytesReference blob = blobs.get(exchange.getRequestURI().getPath());
-                if (blob == null) {
-                    sendError(exchange, RestStatus.NOT_FOUND);
-                    return;
-                }
-                exchange.getResponseHeaders().add("x-ms-blob-content-length", String.valueOf(blob.length()));
-                exchange.getResponseHeaders().add("Content-Length", String.valueOf(blob.length()));
-                exchange.getResponseHeaders().add("x-ms-blob-type", "BlockBlob");
+                final MockAzureBlobStore.AzureBlob azureBlob = mockAzureBlobStore.getBlob(blobPath(exchange), leaseId(exchange));
+
+                final Headers responseHeaders = exchange.getResponseHeaders();
+                final BytesReference blobContents = azureBlob.getContents();
+                responseHeaders.add(X_MS_BLOB_CONTENT_LENGTH, String.valueOf(blobContents.length()));
+                responseHeaders.add("Content-Length", String.valueOf(blobContents.length()));
+                responseHeaders.add(X_MS_BLOB_TYPE, azureBlob.type().getXMsBlobType());
                 exchange.sendResponseHeaders(RestStatus.OK.getStatus(), -1);
 
             } else if (Regex.simpleMatch("GET /" + account + "/" + container + "/*", request)) {
-                // GET Object (https://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectGET.html)
-                final BytesReference blob = blobs.get(exchange.getRequestURI().getPath());
-                if (blob == null) {
-                    sendError(exchange, RestStatus.NOT_FOUND);
-                    return;
-                }
+                // Get Blob (https://learn.microsoft.com/en-us/rest/api/storageservices/get-blob)
+                final MockAzureBlobStore.AzureBlob blob = mockAzureBlobStore.getBlob(blobPath(exchange), leaseId(exchange));
```

Comment:
If we've uploaded blocks, but not the blocklist to complete the upload, we should return a 404 here still.

Reply:
Done in 9fcdd4ce808ba26c7f31e6f91857d35d59818e79